File: 1_two_sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        idx_dict = dict()
        for idx, num in enumerate(nums):
            pair = target-num
            if idx_dict.get(pair) is not None:
                return [idx, idx_dict.get(pair)]
            idx_dict[num] = idx

File: 1_two_sum.go
func twoSum(nums []int, target int) []int {
    m := make(map[int]int)
    for idx, num := range nums {
        if pair_idx, ok := m[target - num]; ok {
            return []int{pair_idx, idx}
        }
        m[num] = idx
    }
    return []int{-1, -1}
}
